cloudinit/dmi.py

```diff
@@ -68,6 +90,29 @@ def _read_dmi_syspath(key):
     return None
 
 
+def _read_kenv(key):
+    """
+    Reads dmi data from FreeBSD's kenv(1)
+    """
+    if key not in DMIDECODE_TO_KENV_MAPPING:
+        return None
+    mapped_key = DMIDECODE_TO_KENV_MAPPING[key]
+
+    LOG.debug("querying dmi data %s", mapped_key)
+
+    try:
+        cmd = ["/bin/kenv", "-q", mapped_key]
```

Comment:
I don't like hard coded paths.
If /bin is not in cloud-init's execution PATH, stuff is probably foobarred.  And if someone modified PATH to put /usr/local/bin in front of /bin and put a kenv executable there, then they probably wanted the one from /usr/local/bin used.

If you want to read a nice smoser rant on this, I've placed one at https://github.com/canonical/cloud-init/pull/584 (I can't figure out how to link to it directly).

Reply:
we have the same policy, for the same reason, in place @ puppet.

so i really don't know why i did this